### ISSUE TITLE
string.Contains returns wrong result on net-ios with .net 9

### ISSUE THREAD

**Issue by markuspalme** [dotnet/runtime#110609]
### Description

`string.Contains` returns unexpected results when called on an empty string on iOS. This is the code in question:

```
"".Contains("ö", StringComparison.CurrentCultureIgnoreCase);
```

Results:

.net 8:

![Image](https://github.com/user-attachments/assets/910136c7-ec14-45d4-be4c-426b1d0c71ca)

.net 9:

![Image](https://github.com/user-attachments/assets/11daf4e8-4cb5-4b57-905c-7dc1eedb9f2c)

Project to reproduce the issue:
https://github.com/markuspalme/dotnetruntime-contains

### Reproduction Steps

To reproduce the issue, call `string.Contains`on an empty string and run the code on `net9-ios`:

```
"".Contains("ö", StringComparison.CurrentCultureIgnoreCase);
```

### Expected behavior

The result should be `false`, an empty string does certainly not contain a character.

### Actual behavior

On .net 9 the call returns `true`.

### Regression?

This is a regression, the same code returns `false` on .net 8.

### Configuration

.net SDK 9.0.100 running on iOS 18.1 ARM64.

**Comment by tarekgh** [comment]
@matouskozak could you please have a look? It looks like the used ICU on iOS does not have all collation data.

**Comment by matouskozak** [comment]
Thanks @markuspalme for filling the issue and @tarekgh for tagging.

I can confirm the issue reproduces. Looks like it works correctly with ASCII search values but breaks with special character. We will have to investigate where the issue is coming from. fyi: @vitek-karas

**Comment by markuspalme** [comment]
This bug not only applies with special characters, but also in this case:

`"".Contains("HW") // returns true`

<img width="965" alt="Image" src="https://github.com/user-attachments/assets/d2ad4b5e-338d-4e48-8aa0-8941e9b2ccfb" />

Given how fundamental `string.contains` is, can the priority of this bug be bumped up? It is causing a lot of issues with filtering down lists etc.

**Comment by matouskozak** [comment]
> This bug not only applies with special characters, but also in this case with ASCII characters:
> 
> `"".Contains("H") // returns true`
> 
> <img alt="Image" width="965" src="https://private-user-images.githubusercontent.com/1481801/413807352-d2ad4b5e-338d-4e48-8aa0-8941e9b2ccfb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk3ODk1ODQsIm5iZiI6MTczOTc4OTI4NCwicGF0aCI6Ii8xNDgxODAxLzQxMzgwNzM1Mi1kMmFkNGI1ZS0zMzhkLTRlNDgtOGFhMC04OTQxZTliMmNjZmIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMTdUMTA0ODA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzhiOWE0N2FiZTEwYzY5NmE2OGQzYTdkYTI5MTRjODA5YTZmMTE2NTdmZDNjN2I5ZjE2Njc5ZDg3MWMzM2E1ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.fM246hUofca6aKSJaHRWihvdPAhXzZqkA7JIlzO_Gaw">
> Given how fundamental `string.Contains` is, can the priority of this bug be bumped up? It is causing a lot of issues with filtering down lists etc.

Yes, you're right. The issue is affecting all iOS scenarios where any culture data is required, i.e., not invariant. The fix is already merged, now it is waiting to be included in future servicing release (probably .NET 9.0.4).

**Comment by matouskozak** [comment]
I just tested the newly released .NET 9.0.4 https://github.com/dotnet/core/blob/main/release-notes/9.0/9.0.4/9.0.4.md and the issue is fixed.

```
"".Contains("ö", StringComparison.CurrentCultureIgnoreCase): False
```

Closing the issue as completed. Please reach out if the issue persists.
